### ISSUE TITLE
Logback shoud default on default JVM charset instead of ASCII

### ISSUE THREAD

**Issue by gbaso** [spring-projects/spring-boot#27230]
Following #23827, default charset where defined for a base logback configuration in 5e26954, as `${CONSOLE_LOG_CHARSET:-default}` and `${FILE_LOG_CHARSET:-default}`.

Unfortunately, logback interprets that as `Charset.forName("default")`, which taking a look at `sun.nio.cs.StandardCharsets$Aliases`  is hardcoded to return ASCII. I feel like `Charset.defaultCharset()` would be a more natural choice.

**Comment by wilkinsona** [comment]
Thanks for the report, @gbaso. Can you please provide a bit more information about when this has caused a problem? `LoggingSystemProperties` should set the `FILE_LOG_CHARSET` and `CONSOLE_LOG_CHARSET` system properties so I'd like to understand exactly when you're seeing the fallback to `default` occurring.

**Comment by gbaso** [comment]
It seems I was mistaken, and my issue was actually related to https://github.com/openjdk/jdk/pull/4733#issuecomment-879856229.

I do, however, still think that spring boot should take a more opinionated approach than defaulting to the "default" charset (which, as I said, is ASCII), using UTF-8 instead.

**Comment by wilkinsona** [comment]
@gbaso When the system properties are in effect, it should already default to UTF-8. To make sure that we fully understand the problem I'd like to know more about your situation and why that apparently wasn't happening. Can you please provide a minimal example that reproduces the problem that you're having?

**Comment by gbaso** [comment]
@wilkinsona I apologize if I was not clear enough, spring boot is working correctly, it was a problem on my end (and in the mechanism the JVM sets the default charset).

However by reading the code, specifically https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot/src/main/resources/org/springframework/boot/logging/logback/defaults.xml, it looks like if for some reasons `FILE_LOG_CHARSET` and `CONSOLE_LOG_CHARSET` are unset, they default to the string `default`:
```xml
<property name="FILE_LOG_CHARSET" value="${FILE_LOG_CHARSET:-default}"/>
```
which is interpreted as `Charset.forName("default")` and is an alias for ASCII.

Given that even OpenJDK is moving to UTF-8 by default, I think we should replace
`value="${FILE_LOG_CHARSET:-default}"` with `value="${FILE_LOG_CHARSET:-UTF-8}"`.

**Comment by wilkinsona** [comment]
> it looks like if for some reasons FILE_LOG_CHARSET and CONSOLE_LOG_CHARSET are unset, they default to the string default

Yes, that's right. What I'd like to understand is why, in your case, those system properties were unset. Can you please provide a minimal example that reproduces that behaviour?

**Comment by gbaso** [comment]
> What I'd like to understand is why, in your case, those system properties were unset

They weren't, I misattributed the source of my issue
